### ISSUE TITLE
tqdm for other languages

### ISSUE THREAD

**Issue by CrazyPython** [tqdm/tqdm#235]
I have made [a list](https://gist.github.com/CrazyPython/dc582f999ebefd7702f7fa5871eab1eb) of tqdm ports. Right now, I think the most important implementations (other than C++) are: (ranked)

 - Java
 - Bash shell
 - C#
 - Objective-C

Not a fan of any of these languages, but that's my opinion.

Note: we should work on a small project to configure tqdm for Jython. It will 2x the time/iteration, though.

**Comment by lrq3000** [comment]
And Julia.

**Comment by CrazyPython** [comment]
@lrq3000 Hm, yes, considering it's used for statistics, where waiting scientists need all teh progress bars.

**Comment by lrq3000** [comment]
In Python, `tqdm` is often used in conjunction with numpy and pandas, so I guess that Julia could also be a good target. At least, when I will start programming with Julia, I will need something like that asap :)

**Comment by CrazyPython** [comment]
@lrq3000 I also added R since it's a widely used statistics lang as well. Often used by first-statisticians then-programmers, they need simplicity.

**Comment by CrazyPython** [comment]
Was doing some research, apparently Ruby is on it's way out for GitHub. (% used) githut.info

**Comment by CrazyPython** [comment]
...and C++ is the language people have the most Issues opened with (only 3rd overall).

**Comment by lrq3000** [comment]
@CrazyPython @casperdcl I created a wiki that is publicly editable to anybody. We can restrict it to repo team members if something bad happens. I created a page out of your list @CrazyPython, you can see it (and modify it) [here](https://github.com/tqdm/tqdm/wiki/tqdm-ports-in-other-languages) :)

Thank you for your list @CrazyPython ! Also feel free to add the least popular reimplementation of tqdm, it's good to know they exist (they can have good exclusive features).

**Comment by lrq3000** [comment]
@CrazyPython yes agree about R, but I think it's also going out of favor too in favor of Octave/Matlab, Python and Julia in the far future (I can hope :) ). But still if we can support it that would be cool.

**Comment by CrazyPython** [comment]
@lrq3000 spent a lot of time on that formatting there. great to know it carried over! took screenshots of flairs to try to make some images but that didn't work. have the "Official" and "Pre-alpha" labels in my `tqdm-rsrcs` repo.

**Comment by CrazyPython** [comment]
I think the > symbol looks better with monospacing... what do you think @lrq3000

**Comment by lrq3000** [comment]
Also a Javascript version would be cool, similar to the node tqdm version, maybe we could merge both projects.

@CrazyPython Yes but that was weird to have a link on it, however we can put monospacing outside. But then it's a bit weird to have monospacing for everything, it looks like some sort of code or quote. So I don't know XD I'll leave it up to you @CrazyPython !

**Comment by CrazyPython** [comment]
@lrq3000 Here's a visually pleasing alternative:

https://github.com/tqdm/tqdm/wiki/tqdm-ports-in-other-languages

**Comment by CrazyPython** [comment]
Open source development in an instrumental song: https://www.youtube.com/watch?v=3I_n7it4hKw

**Comment by lrq3000** [comment]
Ok good for me :) About the label, thank you for making them, we can maybe use them but I'm not sure that's necessary for now (because there are only two labels in the current wiki XD).

From githut.info (which I didn't know before):

> Programming languages are not simply the tool developers use to create programs or express algorithms but also instruments to code and decode creativity. By observing the history of languages we can **enjoy the quest of human kind** for a better way to solve problems, to facilitate collaboration between people and to reuse the effort of others.

So epic! XD

**Comment by CrazyPython** [comment]
@lrq3000 Yup. A story on the monkey-patchy way I made them:

1. Reddit flairs matched what I wanted.
2. So I found one of my favorite small subreddits (/r/ScenesFromAHat, comedy subreddit I read and post in - check it out, it's funny!) that had rounded flairs.
3. Inspect Element > Change colors
4. Screenshot
5. Crop in Preview.app

There are no artifacts in the image - it's correct down to the pixel. So I guess my bodgy way worked!

**Comment by lrq3000** [comment]
Arg yeah but then I'm not sure the licensing would allow us to reuse that :/

**Comment by CrazyPython** [comment]
@lrq3000 It's a tiny bit of CSS... and reddit is open source. I have good rels with the mods of /r/ScenesFromAHat so I'll go off to contact them.

**Comment by lrq3000** [comment]
It should be under MIT, but anyway it's not hard to make a new one out of nothing (even if uglier, at least it'll be compatible).

**Comment by CrazyPython** [comment]
*tl;dr* basically we can write tqdm ports using stuff like Jython and R integration

Rewriting tqdm for an entirely new language is hard. But we can port tqdm to e.g. Java and R much more easily with stuff like Jython and R's built in Pythonintegration.

Jython layout:

tqdm python (stripped down without unportable features)
↓
Jython
↓
Small Java tqdm handler to make the API simpler

This may be slower, but it's better than nothing and much easier than rewriting the entirety of tqdm. It also lets us do this:

💻 📈🌏😈

**Comment by samssann** [comment]
@CrazyPython you are pretty much done with R if you get the C++ port working (https://github.com/RcppCore/Rcpp).
